iib/workers/tasks/git_utils.py

```diff
@@ -0,0 +1,255 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+"""This file contains functions for saving changes to Git."""
+import logging
+import os
+import tempfile
+import shutil
+from typing import Optional, Tuple
+
+from operator_manifest.operator import ImageName
+
+from iib.common.tracing import instrument_tracing
+from iib.exceptions import IIBError
+from iib.workers.config import get_worker_config
+from iib.workers.tasks.utils import run_cmd
+
+
+log = logging.getLogger(__name__)
+
+
+@instrument_tracing(span_name="workers.tasks.git_utils.push_configs_to_git")
+def push_configs_to_git(
+    request_id: int, from_index: str, src_configs_path: str, commit_message: Optional[str] = None
+) -> None:
+    """
+    Pushes /configs subfolders to a Git repository.
+
+    :param int request_id: The ID of the IIB request.
+    :param str from_index: The from_index pullspec. Note: This should have a tag that
+                           corresponds to the OCP version, like 'v4.19'. The code
+                           assumes that the branch already exists in remote repo.
+    :param str src_configs_path: Path to /configs folder where <pkg>/catalog.json
+                           files reside.
+    :param str commit_message: Custom commit message. If None, a default message is used.
+    :raises IIBError: If src_configs_path is not found, remote branch does not exist,
+                      or a Git operation fails.
+    """
+    # Determine Git repo from pullspec
+    index_image = ImageName.parse(from_index)
+    branch = index_image.tag
+    repo_url = get_gitlab_url(from_index)
+    gitlab_token_name, gitlab_token = get_gitlab_token(repo_url)
```

Comment:
@lipoja Would you prefer get_git_url / get_git_token here? 
I think that GitHub could be used here as well (mapped GitHub URL + GitHub token).

Reply:
Originally I kept "gitlab" in the name since we have the `iib_index_configs_gitlab_tokens_map` and `iib_web_index_to_gitlab_push_map` configs. So we may want to rename those too in some future story. But I changed the method names in this PR per your suggestion.